src/main/java/io/github/connorhartley/guardian/sequence/SequenceController.java

```diff
@@ -78,7 +78,12 @@ public void invoke(Player player, Event event) {
                 return false;
             }
 
-            SequenceFinishEvent attempt = new SequenceFinishEvent(sequence, player, sequence.getSequenceReport(), Cause.of(NamedCause.source(this.plugin), NamedCause.of("CONTEXT", sequence.getContext())));
+            if (sequence.hasStarted()) {
+                sequence.getContextHandler().stop();
+            }
+
+            SequenceFinishEvent attempt = new SequenceFinishEvent(sequence, player, sequence.getSequenceReport(),
+                    Cause.of(NamedCause.source(this.plugin), NamedCause.of("CONTEXT", sequence.getContextContainer())));
```

Comment:
I'm curious if Guardian should create its own cause for these sequence events or is that unnecessary?